Lib/fractions.py

```diff
@@ -247,12 +248,23 @@ def limit_denominator(self, max_denominator=1000000):
             n, d = d, n-a*d
 
         k = (max_denominator-q0)//q1
-        bound1 = Fraction(p0+k*p1, q0+k*q1)
-        bound2 = Fraction(p1, q1)
-        if abs(bound2 - self) <= abs(bound1-self):
-            return bound2
+        bound1_n = p0 + k*p1
+        bound1_d = q0 + k*q1
+        bound2_n = p1
+        bound2_d = q1
+        bound1_minus_self_n = abs((bound1_n * d_original)
+                                  - (n_original * bound1_d))
+        bound1_minus_self_d = d_original * bound1_d
+        bound2_minus_self_n = abs((bound2_n * d_original)
+                                  - (n_original * bound2_d))
+        bound2_minus_self_d = d_original * bound2_d
+
+        difference = ((bound1_minus_self_n * bound2_minus_self_d)
+                      - (bound2_minus_self_n * bound1_minus_self_d))
+        if difference >= 0:
+            return Fraction(bound2_n, bound2_d)
```

Comment:
As you suggested in your comments, it's safe to use `_normalize=False` here and below: you can verify by induction that `p0 * q1 - p1 * q0` is `±1` at the beginning and the end of each iteration of the `while` loop (the sign alternates on each iteration), and it follows that `gcd(p0 + k*p1, q0 + k*q1) = 1` and that `gcd(p1, q1) = 1`.

Reply:
That's what I thought too! but didn't want to suggest a change beyond the scope of the Issue/PR.